server/src/main/kotlin/com/kamelia/jellyfish/rest/user/Routes.kt

```diff
@@ -21,6 +25,25 @@ private fun Route.signup() = postOrCatch<UserDTO> { body ->
     call.respond(UserService.signup(body))
 }
 
+private fun Route.getUserById() = getOrCatch(path = "/{uuid}") {
+    val uuid = call.getUUID()
+    call.respond(UserService.getUserById(uuid))
+}
+
+private fun Route.getAllUsers() = getOrCatch {
+    call.respond(UserService.getUsers())
+}
+
+private fun Route.getPagedUsers() = getOrCatch(path = "/paged") {
+    val page = (call.request.queryParameters["page"] ?: "0").let {
+        it.toLongOrNull() ?: throw IllegalArgumentException("Invalid page number")
+    }
+    val pageSize = (call.request.queryParameters["pageSize"] ?: "25").let {
+        it.toIntOrNull() ?: throw IllegalArgumentException("Invalid page size")
+    }
+    call.respond(UserService.getUsers(page, pageSize))
+}
```

Comment:
Why did you split this?

Reply:
Because some admin user (or an alternate api consumer) might want to get every user at some point. There are several solutions to this actually, like requesting a page once, getting the total, and getting a new page of this size, but this is not ideal. Another solution that is not the one above was to say that if pageSize == -1, then get all users, but i find this less than ideal in terms of clarity, if that makes sense

Reply:
Ok, so I suggest we use the `/api/users` route to work with `page` and `pageSize` and have an additional `/api/users/all` route.

Reply:
fair enough